oeps/oep-0028-content-theming-in-react.rst

```diff
@@ -0,0 +1,246 @@
+
+.. code-block:: js
+
+    class _Header extends React.PureComponent {
+        render() {
+            return (
+                <header>
+                    <SiteLogo/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+
+* We will follow a 2 step approach to create any UI Component. We will first break down a complex component into smaller non-customizable components. We will then compose these smaller components to form a complex UI.
+
+We can see this through an example where we have a theme, which has several components for a default theme.
+
+.. code-block:: js
+
+    // Customizable Header
+    class _Header extends React.PureComponent {
+        render() {
+            return (
+                <header>
+                    <SiteLogo/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+
+    // Customizable Main Navigation Area
+    class _MainNav extends React.PureComponent {
+        render() {
+            return (
+                <MainNavWrapper>
+                    <a href="/">Home</a>
+                    <LoginLink/>
+                    {this.extraNavLinks}
+                </MainNavWrapper>
+            );
+        }
+        get extraNavLinks() { return []; }
+    }
+    // Internal MainNavWrapper - not meant to be modified in most cases
+    class _MainNavWrapper extends React.PureComponent {
+        render() {
+            return (
+                <div className="mainNav">
+                    <ul>
+                        {React.Children.map(this.props.children, (child) => (child ? <li>{child}</li> : null))}
+                    </ul>
+                </div>
+            )
+        }
+    }
+
+    // Default Theme:
+    export const Header = _Header;
+    export const MainNav = _MainNav;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+Now if we want to customize our _Header component, we can easily do it like
+
+.. code-block:: js
+
+    // SiteLogo being updated in Header
+    class MyThemedHeader extends _Header {
+        render() {
+            return (
+                <header>
+                    {/* Replace <SiteLogo/> with a fancy widget */}
+                    <MyCustomAnimatedLogoWidget/>
+                    <MainNav/>
+                    <UserAvatar/>
+                </header>
+            );
+        }
+    }
+    // Custom theme:
+    export const Header = MyThemedHeader;
+    export const MainNavWrapper = _MainNavWrapper;
+
+
+* We will provide support to add data or other components by using `props.children`. The `props.children` property is provided from React and can be extended to include specialized children like left and right for a Pane component. An example of this can be
```

Comment:
I'm not sure that this paragraph adds any useful information since it's just describing a core react feature?

>The `props.children` property is provided from React and can be extended to include specialized children like left and right for a Pane component.

Nit: If you want do do a `left` and `right` pane component, those props are not extensions of the `props.children` property. The `props.children` property cannot be "extended" in any meaningful sense. What you are describing is just using other properties in a similar way.

Reply:
I have removed the point where `props.children` and `props.left/right` were being mentioned. My reason to remove is as follows
* These are standard react features, anyone developing in React must be already familiar with it
* We are not doing anything additional from the standard convention.